BMSUtils.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <assert.h>
+#include "BMSUtils.h"
+
+
+Parameter::Parameter(Language Lang, float min_value, float max_value, string name)
+{
+    min_threshold = min_value;
+    max_threshold = max_value;
+    parameter_name = name;
+    tolerance = max_threshold * 0.05 ;
+    if(Lang == ENGLISH){
+
+        error_msg = "Out of range";
+        warning_msg = "Close to warning range";
+
+    }
+    else if(Lang == DEUTSCH){
+        error_msg = "Außer Reichweite";
+        warning_msg = "In der Nähe des Warnbereichs";
+    }
+}
+
+bool Parameter :: check_min(float value){
+    if(value>=min_threshold)
+    {
+        if(value <= min_threshold + tolerance)
+            cout << parameter_name << " - "<< warning_msg <<endl;
+        return true;
+    }
+    else
+    {
+        cout << parameter_name << " - "<< error_msg <<endl;
+        return false;
+             
+    }
+       
+ 
+};
+
+bool Parameter ::  check_max(float value){
+
+    if(value<=max_threshold)
+        {
+            if(value >= max_threshold - tolerance)
+                cout << parameter_name << " - "<< warning_msg <<endl;
+            return true;
+        }
+        else
+        {
+                cout << parameter_name << " - "<< error_msg <<endl;
+                return false;
+        }
+
+};
+
+bool Parameter::check_threshold(float value){
+
+    return (check_min(value) &&  check_max(value));
+};
+
+void temperature::Conversion (UnitType unit){
+
+    if (unit == FAHRENHEIT){
+         min_threshold = (min_threshold - 32)*5/9;
+         max_threshold = (max_threshold - 32)*5/9;
+    }
+
+};
+
+temperature::temperature(Language Lang, float min_value, float max_value, string name ,UnitType unit) 
+: Parameter (Lang, min_value, max_value, name ) 
+{
+    Conversion(unit);
```

Comment:
Nice initialization in constructor. The unit-logic stays encapsulated in the class.